ua_parser/user_agent_parser.py

```diff
@@ -218,8 +219,15 @@ def Parse(self, user_agent_string):
 MAX_CACHE_SIZE = 200
 _PARSE_CACHE = {}
 
+_UA_TYPES = str
+if sys.version_info < (3,):
+    _UA_TYPES = (str, unicode)
+
 
 def _lookup(ua, args):
+    if not isinstance(ua, _UA_TYPES):
+        raise TypeError("Expected user agent to be a string, got %r" % ua)
```

Comment:
shouldn't that be `type(ua)` at the end?

Reply:
Shouldn't really matter, this prints the value which I assumed would be more relevant / understandable.